crates/mockiato-codegen/src/generate/arguments.rs

```diff
@@ -83,33 +78,16 @@ fn generate_display_impl(method_decl: &MethodDecl, generics: &TokenStream) -> To
     }
 }
 
-/// Generates a `Debug` implementation for an arguments struct.
-fn generate_debug_impl(method_decl: &MethodDecl, generics: &TokenStream) -> TokenStream {
-    let arguments_ident = arguments_ident(&method_decl.ident);
-    let arguments_ident_str_literal = ident_to_string_literal(&arguments_ident);
-
-    let debug_fields: TokenStream = method_decl
-        .inputs
-        .args
-        .iter()
-        .map(|input| {
-            let ident = &input.ident;
-            let ident_as_str = ident_to_string_literal(ident);
-            quote! {
-                .field(#ident_as_str, &mockiato::internal::MaybeDebugWrapper(&self.#ident))
-            }
-        })
-        .collect();
-
-    quote! {
-        impl #generics std::fmt::Debug for #arguments_ident #generics {
-            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-                f.debug_struct(#arguments_ident_str_literal)
-                 #debug_fields
-                 .finish()
-            }
-        }
-    }
+fn debug_impl_fields<'a>(
+    method_decl: &'a MethodDecl,
+) -> impl Iterator<Item = DebugImplField<'a>> + 'a {
+    method_decl.inputs.args.iter().map(|input| {
+        let ident = &input.ident;
```

Comment:
The following lines are a bit weird, I don't understand what's happening. Why are we returning a tuple?

Reply:
Addressed in eeb2fa8